docs/development/systemtask.md

```diff
@@ -0,0 +1,263 @@
+> algorithmic core as a library that does not depend on FreeRTOS- and
+> ARM-related headers, and calling this library from your task.
+> This makes it easier to write unit tests for your algorithmic core
+> that can be compiled, run, and debugged on a PC.
+
+
+Writing the public interface
+----------------------------
+
+Subsystems should expose a public interface to other parts of the firmware
+that hide implementation details as much as is practical.
+We create the new file in `modules/interface/example.h`:
+
+``` {.c}
+#ifndef __EXAMPLE_TASK_H__
+#define __EXAMPLE_TASK_H__
+
+#include <stdbool.h>
+
+void exampleTaskInit();
+bool exampleTaskTest();
+
+void exampleTaskEnqueueInput(int value);
+
+#endif // __EXAMPLE_TASK_H__
+```
+
+In a real task, make sure to comment the public API thoroughly.
+
+
+Initializing the task
+---------------------
+
+In `modules/src/system.c`, make the following changes.
+
+Include the header for our new task:
+
+``` {.c}
+#include "example.h"
+```
+
+
+In `systemTask()`, after the lines where the other `Init()` functions
+are called, add the line
+
+``` {.c}
+exampleTaskInit();
+```
+
+
+In `systemTask()`, after the lines where the other `pass &= ...Test()` lines,
+add the line
+``` {.c}
+pass &= exampleTaskTest();
+```
+
+
+Adding the task to the build
+----------------------------
+
+Add this to the Makefile, after the end of the `Modules` block:
+
+``` {.make}
+PROJ_OBJ += example.o
+```
+
+
+Compile, flash and run!
+-----------------------
+
+Now the last step is to compile and flash your new firmware. Launch the
+following commands in a shell:
+
+``` {.bash}
+crazyflie-firmware$ make
+crazyflie-firmware$ make cload
+```
+
+The output will be similar to the following:
+
+``` {.bash}
+crazyflie-firmware$ make
+(...)
+  CC    hello.o
+(...)
+Crazyflie 2.0 build!
+Build 44:61f3d878233d (2015.08.1-44) MODIFIED
+Version extracted from git
+Crazyloader build!
+   text    data     bss     dec     hex filename
+ 130660    1636   29828  162124   2794c cf2.elf
+rm version.c
+crazyflie-firmware$ make cload
+../crazyflie-clients-python/bin/cfloader flash cf2.bin stm32-fw
+Restart the Crazyflie you want to bootload in the next
+ 10 seconds ...
+ done!
+Connected to bootloader on Crazyflie 2.0 (version=0x10)
+Target info: nrf51 (0xFE)
+Flash pages: 232 | Page size: 1024 | Buffer pages: 1 | Start page: 88
+144 KBytes of flash available for firmware image.
+Target info: stm32 (0xFF)
+Flash pages: 1024 | Page size: 1024 | Buffer pages: 10 | Start page: 16
+1008 KBytes of flash available for firmware image.
+
+Flashing 1 of 1 to stm32 (fw): 161867 bytes (159 pages) ..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10.........9
+Reset in firmware mode ...
+$
+```
+
+Since our task doesn't do anything, there is nothing to observe when running
```

Comment:
If you want to you could add a DEBUG_PRINT() to the new task to show that it is running

Reply:
Added in d3c0a97.